### PR TITLE
Divider fixes and docs

### PR DESCRIPTION
**Pull request by dbanksdesign**
*Issue #, if available:*

*Description of changes:*

* Updated divider docs to use new docs components and example infrastructure
* Fixed orinetation vertical issue
* Added design tokens for more customization

https://user-images.githubusercontent.com/321279/142040023-5c9e628d-efec-43c6-a583-9d86752c51a5.mp4

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/src/pages/components/divider/DividerPropControls.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import React from 'react';
```
Comment (zchenwei) [inline docs/src/pages/components/divider/DividerPropControls.tsx:1]:
```suggestion
import * as React from 'react';
```

File: docs/src/components/useGridContainerProps.tsx [hunk 2]
```diff
@@ -1,6 +1,5 @@
 import { DividerOptions, GridContainerStyleProps } from '@aws-amplify/ui-react';
 import { useState } from 'react';
```
Comment (zchenwei) [inline docs/src/components/useGridContainerProps.tsx:2]:
```suggestion
import * as React from 'react';

import { DividerOptions, GridContainerStyleProps } from '@aws-amplify/ui-react';
```

File: docs/src/pages/components/divider/DividerPropControls.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import { DividerOptions, Flex, SelectField } from '@aws-amplify/ui-react';
+import { FieldLabeler } from '../../../components/FieldLabeler';
+import { DemoBox } from '../../../components/DemoBox';
+export interface DividerPropControlsProps extends DividerOptions {
```
Comment (zchenwei) [inline docs/src/pages/components/divider/DividerPropControls.tsx:5]: a line break

File: docs/src/pages/components/divider/examples/DividerClassNameExample.tsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { Flex, Divider } from '@aws-amplify/ui-react';
+
+const css = `.custom-divider {
+  border-style: dashed;
+}`;
+
+export const DividerClassNameExample = () => (
+  <>
```
Comment (zchenwei) [inline docs/src/pages/components/divider/examples/DividerClassNameExample.tsx:8]: Looks like this `<>` is not necessary

File: docs/src/pages/components/divider/examples/DividerThemeExample.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    },
+    components: {
+      divider: {
+        borderStyle: { value: 'dotted' },
+        // You can reference other theme tokens:
+        borderColor: { value: '{colors.brand.secondary.80.value}' },
+      },
+    },
+  },
+};
```
Comment (zchenwei) [inline docs/src/pages/components/divider/examples/DividerThemeExample.tsx:18]: 👏

**Review by zchenwei — APPROVED** [review]
The example code is looking great!!! No blockers that I see

**Review by hvergara — APPROVED** [review]

**Pull request merged**